### PR TITLE
feat: Add Flank GitHub action

### LINKED ISSUE #1596: Download Flank in Github Action
Download Flank in new created Github action with the version specified by the user or latest if not present.
Add it to the path allowing using it in the next steps

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1596 

### Flank scripts part
Run command
- `flankScripts github download_flank` -> to get latest version
- `flankScripts github download_flank <version>` -> to get specified version

### Github action part
You could run Flank GitHub action on [testing repository](https://github.com/Flank/flank-github-action-testing/actions)

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: flank-scripts/src/main/kotlin/flank/scripts/cli/github/DownloadFlankCommand.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package flank.scripts.cli.github
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.option
+import flank.scripts.ops.github.downloadFlank
+import kotlinx.coroutines.runBlocking
+
+object DownloadFlankCommand : CliktCommand(
+    name = "downloadFlank",
```
Comment (piotradamczyk5) [inline flank-scripts/src/main/kotlin/flank/scripts/cli/github/DownloadFlankCommand.kt:9]:
```suggestion
    name = "download_flank",
```

File: action.yml [hunk 2]
```diff
@@ -1,26 +1,19 @@
...
+  steps:
+    - name: Download flankScripts
+      run: |
+        curl -L https://github.com/Flank/flank/releases/download/flank-scripts-1.7.3/flank-scripts.jar --output flank-scripts.jar
+        echo 'java -jar ./flank-scripts.jar "$@"' > flankScripts
+        chmod +x flankScripts
+      shell: bash
+    - id: download_flank
+      run: |
+        ./flankScripts downloadFlank --version=${{ inputs.version }}
```
Comment (piotradamczyk5) [inline action.yml:18]:
```suggestion
        ./flankScripts github download_flank --version=${{ inputs.version }}
```

File: flank-scripts/build.gradle.kts [hunk 3]
```diff
@@ -26,7 +26,7 @@ shadowJar.apply {
     }
 }
 // <breaking change>.<feature added>.<fix/minor change>
-version = "1.7.3"
+version = "1.7.4"
```
Comment (piotradamczyk5) [inline flank-scripts/build.gradle.kts:29]:
```suggestion
version = "1.8.0"
```

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+inputs:
+  version: 
+    description: 'Flank version to run'
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Download flankScripts
+      run: |
+        curl -L https://github.com/Flank/flank/releases/download/flank-scripts-1.7.3/flank-scripts.jar --output flank-scripts.jar
```
Comment (pawelpasterz) [inline action.yml:12]: Do we want to have `1.7.3` hardcoded?

Comment (piotradamczyk5) [reply action.yml:12]: a version of Flank scripts will be hardcoded to use the tested and stable version for Github action.
We will update it manually

Comment (pawelpasterz) [reply action.yml:12]: Makes sense 👍  Thanks

Comment (pawelpasterz) [reply action.yml:12]: But should we use `1.8.0` then?

Comment (piotradamczyk5) [reply action.yml:12]:
```suggestion
        curl -L https://github.com/Flank/flank/releases/download/flank-scripts-1.8.0/flank-scripts.jar --output flank-scripts.jar
```

**Review by Sloox — APPROVED** [review]

**Review by pawelpasterz — APPROVED** [review]

**Pull request merged**